src/qibosoq/programs/base.py

```diff
@@ -244,23 +239,23 @@ def acquire(
         # if there are no readouts, temporaray set 1 so that qick can execute properly
         reads_per_rep = 1 if readouts_per_experiment == 0 else readouts_per_experiment
 
-        res = super().acquire(
+        res = self.acquire(  # pylint: disable=E1123
```

Comment:
Actually, Pylint seems right in flagging this as an error, why did you disable?

https://github.com/openquantumhardware/qick/blob/a17f2dbc449bfcc723da33795bac3d4fc11cba25/qick_lib/qick/qick_asm.py#L1454

Reply:
Pylint see an error because it thinks that QickProgram.acquire will be called, actually we always call AveragerProgram.acquire (or NDAverager) that has the readouts_per_experiment agument

https://github.com/openquantumhardware/qick/blob/a17f2dbc449bfcc723da33795bac3d4fc11cba25/qick_lib/qick/averager_program.py#L71

Reply:
I actually thought the same as Pylint: then why do we inherit from `QickProgram`, and not from `AveragerProgram`?

Reply:
Because the base program is used both by the pulse sequence and the sweeper program and they are, respectively, a AveragerProgram and a NDAveragerProgram. Both inheriting from QickProgram

Reply:
This issue is just caused by the fact that there is this change of name in the function parameters. If this was not the case, I think this would have been perfectly fine... because of that I have added the pylint exception

Reply:
> (it would be even interesting to know why they have a different and incompatible interface...)
My guess: to be completely sure of which function they are actually using...

Reply:
Maybe it's worth to open a tiny issue on Qick.

If they are inheriting, and the base program is establishing an interface, it would be nice to preserve it.

Actually, they could have used different methods names (to signal that they are different operations). And this would have been fine from their side.
In which case we could not use just `self.acquire`, and we might have had to distinguish between `Averager` and `NDAverager`.

What we could do is to implement an `BaseProgram._acquire`, that does everything but the call to the underlying `acquire` function, and implement this in the subclasses

Reply:
> Maybe it's worth to open a tiny issue on Qick.

At the moment qibosoq is not compatible with the newest versions of qick, so I had avoided it... I don't think it's really worth it honestly, but sure I may be wrong

> What we could do is to implement an `BaseProgram._perform_experiment`, that does everything but the call to the underlying `acquire` function, and implement this in the subclasses.

I have though of that, but in any case I will end up complicating the code and duplicating some parts

Reply:
> I have though of that, but in any case I will end up complicating the code and duplicating some parts

It's not urgent, but we could solve this problem, e.g. with a generator (in the general case, you can use a generator, i.e. a function that `yield`s, also to pause the execution and *receive* a value, on top of giving back one to the caller).

I believe we could solve in an elegant enough way, without duplication. But it's definitely not urgent, so please, turn this into an issue (if it will age too much, and we won't care, we'll then close the issue).